openslides_backend/action/actions/user/account_import.py

```diff
@@ -0,0 +1,174 @@
+
+        if self.import_state != ImportState.ERROR:
+            create_action_payload: List[Dict[str, Any]] = []
+            update_action_payload: List[Dict[str, Any]] = []
+            self.flatten_object_fields(["username", "saml_id", "default_password"])
+            for row in self.rows:
+                if row["state"] == ImportState.NEW:
+                    create_action_payload.append(row["data"])
+                else:
+                    update_action_payload.append(row["data"])
+            if create_action_payload:
+                self.execute_other_action(UserCreate, create_action_payload)
+            if update_action_payload:
+                self.execute_other_action(UserUpdate, update_action_payload)
+
+        return {}
+
+    def validate_entry(self, row: ImportRow) -> ImportRow:
+        entry = row["data"]
+        username = self.get_value_from_union_str_object(entry.get("username"))
+        if not username:
+            raise ActionException(
+                "Invalid JsonUpload data: The data from json upload must contain a valid username object"
+            )
+        check_result = self.username_lookup.check_duplicate(username)
+        id_ = cast(int, self.username_lookup.get_field_by_name(username, "id"))
+
+        if check_result == ResultType.FOUND_ID and id_ != 0:
+            if row["state"] != ImportState.DONE:
+                row["messages"].append(
+                    f"Error: row state expected to be '{ImportState.DONE}', but it is '{row['state']}'."
+                )
+                row["state"] = ImportState.ERROR
+                entry["username"]["info"] = ImportState.ERROR
+            elif "id" not in entry:
+                raise ActionException(
+                    f"Invalid JsonUpload data: A data row with state '{ImportState.DONE}' must have an 'id'"
+                )
+            elif entry["id"] != id_:
+                row["state"] = ImportState.ERROR
+                entry["username"]["info"] = ImportState.ERROR
+                row["messages"].append(
+                    f"Error: username '{username}' found in different id ({id_} instead of {entry['id']})"
+                )
+        elif check_result == ResultType.FOUND_MORE_IDS:
+            row["state"] = ImportState.ERROR
+            entry["username"]["info"] = ImportState.ERROR
+            row["messages"].append(
+                f"Error: username '{username}' is duplicated in import."
+            )
+        elif check_result == ResultType.NOT_FOUND_ANYMORE:
+            row["messages"].append(
+                f"Error: user {entry['username']['id']} not found anymore for updating user '{username}'."
+            )
+            row["state"] = ImportState.ERROR
+        elif check_result == ResultType.NOT_FOUND:
+            pass  # cannot create an error !
+
+        saml_id = self.get_value_from_union_str_object(entry.get("saml_id"))
+        if saml_id:
```

Comment:
You could deduplicate the `saml_id` section with the `username` section since it's mostly the same checks

Reply:
IMO they are similar, because both are checking the duplicate state, but the existing differences are the important thing. I prefer to let this parts separated.

Reply:
You coould incoorporate the difference in the deduplication, but okay for me